### PR TITLE
distribution/publishing: Add debs to releases

### LINKED ISSUE #29956: Publishing debs with releases
We have an existing pipeline for producing tested debs which i am seeking to enable

To sign the debs in such a way that they can be consumed by an apt repository (reprepro) we need to sign changes files for any named distributions we want to include the debs for.

When we multiply variants (eg envoy/envoy-contrib/perhaps dbg) by distros by architectures, this creates a lot of changes files and would potentially make it harder to (humanly) discern downloadables

One possible solution for this im pondering is to put all the debian binaries and associated changes files into a tarball that any repos could cosume, but would not be such a release hog

### PR DESCRIPTION
**Pull request by phlax**
Fix #29956 

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
currently this is missing contrib debs - which we can add but it multiplies the number

there are double the number of files i would like - it was done this way to meet the expectations of reprepro to allow us to set up an apt repo - but its not the only repo builder, not sure if others would require less changes files

**Comment by phlax** [comment]
checking further - i think the `.changes` are all needed

File: distribution/sign.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -e
+
```
Comment (phlax) [inline distribution/sign.sh:2]: add some explanation about how signing tool works and/or do some of the path mangling bazel side

**Review by wbpcode — APPROVED** [review]
LGTM.

**Pull request merged**